scvi/model/utils/_latent.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from anndata import AnnData
+from scipy.sparse import csr_matrix
+
+from scvi import REGISTRY_KEYS
+from scvi._types import LatentDataType
+from scvi.data._constants import _ADATA_LATENT_UNS_KEY
+from scvi.data.fields import NumericalObsField, ObsmField, StringUnsField
+
+
+def scvi_get_latent_adata_from_adata(
+    adata: AnnData,
+    mode: LatentDataType,
+    scvi_latent_qzm_key: str,
+    scvi_latent_qzv_key: str,
+    scvi_obs_lib_size_key: str,
+    use_latent_qzm_key: str = "X_latent_qzm",
+    use_latent_qzv_key: str = "X_latent_qzv",
+):
+    """TODO add docstring"""
+    if mode == "dist":
+        adata.obsm[scvi_latent_qzm_key] = adata.obsm[use_latent_qzm_key]
+        adata.obsm[scvi_latent_qzv_key] = adata.obsm[use_latent_qzv_key]
+        adata.obs[scvi_obs_lib_size_key] = np.squeeze(np.asarray(adata.X.sum(axis=1)))
```

Comment:
this currently does not handle the case in which a user passed a size factor key to setup anndata

Reply:
in which case you'd have to get that from the anndata manager

Reply:
ok actually this is confusing code in the way vae is structured. I feel like we should lightly refactor the pytorch module to have a simpler control flow....

Reply:
the simple hack would be to check if the user has set the size factor key, and if they haven't, compute the library size and store it in our existing size factor key.

then the cached inference would return a dummy library which would actually be unused.